crates/fetch_queue/Cargo.toml

```diff
@@ -0,0 +1,17 @@
+[package]
+name = "kitsune2_fetch_queue"
+version = "0.0.1-alpha"
+description = "queue for fetching kitsune ops from peers"
+license = "Apache-2.0"
+homepage = "https://github.com/holochain/kitsune2"
+documentation = "https://docs.rs/kitsune2_fetch_queue"
+authors = ["Holochain Core Dev Team <devcore@holochain.org>"]
+keywords = ["holochain", "holo", "p2p", "dht", "networking"]
+categories = ["network-programming"]
+edition = "2021"
+
+[dependencies]
+kitsune2_api = { path = "../api", version = "0.0.1-alpha" }
+bytes = { workspace = true }
+rand = "0.8.5"
+tokio = { workspace = true, features = ["full"] }
```

Comment:
Yes, if this is starting to be used as a real runtime dependency, we need to figure out what this looks like.

Perhaps it is naive of me to say I'd like kitsune to be async runtime independent...

I *think*  from tokio or some other runtime, we only care about spawning tasks and async timeouts. Both of these we could abstract with a trait that the implementer could fill out if they want, and we could provide a "tokio" feature that would provide a default impl of said trait through tokio.

Everything else (channels, etc) we could get from the futures crate.

It is a given that this approach would be more overhead for us as developers. And currently tx5 internally uses tokio directly, so for this to be useful, we'd have to gut that as well.

Is the end result of this worth the effort to us as an org?

Reply:
Actually having tokio with full features in the dependencies instead of the dev-dependencies was a mistake I forgot to fix.

Regarding async runtime-agnosticism, I'm up for evaluating how much overhead this is to adhere to. In general I don't see a pull for it, as I don't even know how much potential adoption kitsune could have outside of Holochain.